File: summer2019/week1/plus-one.java
class Solution {
    public int[] plusOne(int[] digits) {
        if (digits[digits.length - 1] < 9) {
            digits[digits.length - 1] += 1;
            return digits;
        }
        int[] output = new int[digits.length];
        int iterator = digits.length - 1;
        int carry = 1;
        while (iterator >= 0) {
            int added = digits[iterator] + carry;
            carry = 0;
            if (added >= 10) {
                carry = 1;
                added = 0;
            }
            output[iterator] = added;
            iterator--;
        }
        if (carry != 0) {
            int[] result = new int[output.length + 1];
            result[0] = carry;
            System.arraycopy(output, 0, result, 1, output.length);
            return result;
        }
        return output;
    }
}

File: summer2021/unit1/plus-one.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        result = []
        carry = 1
        for i in digits[::-1]:
            add_one = i + carry
            if add_one >= 10:
                carry = 1
                result.insert(0, 0)
            else:
                result.insert(0, add_one)
                carry = 0
        if carry > 0:
            result.insert(0, carry)
        return result